contracts/axelarnet-gateway/src/executable.rs

```diff
@@ -0,0 +1,81 @@
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{HexBinary, WasmMsg};
+use router_api::{Address, CrossChainId};
+
+#[cw_serde]
+pub enum AxelarExecutableMsg {
+    /// Execute the message at the destination contract with the corresponding payload, via the gateway.
+    Execute {
+        cc_id: CrossChainId,
+        source_address: Address,
+        payload: HexBinary,
+    },
+}
+
+impl<'a> From<client::Client<'a, AxelarExecutableMsg, ()>> for AxelarExecutableClient<'a> {
+    fn from(client: client::Client<'a, AxelarExecutableMsg, ()>) -> Self {
+        AxelarExecutableClient { client }
+    }
+}
+
+pub struct AxelarExecutableClient<'a> {
```

Comment:
the client is only relevant to the gateway itself (since it executes it). Only the ExecuteMsg needs to be public for apps to implement it